src/dotty/tools/dotc/repl/CompilingInterpreter.scala

```diff
@@ -383,24 +387,54 @@ class CompilingInterpreter(out: PrintWriter, ictx: Context) extends Compiler wit
       names1 ++ names2
     }
 
+    /** Sets both System.{out,err} and Console.{out,err} to supplied
+     *  `os: OutputStream`
+     */
+    private def withOutput[T](os: ByteOutputStream)(op: ByteOutputStream => T) = {
+      val ps     = new PrintStream(os)
+      val oldOut = System.out
+      val oldErr = System.err
+      System.setOut(ps)
+      System.setErr(ps)
+
+      try {
+        Console.withOut(os)(Console.withErr(os)(op(os)))
+      } finally {
+        System.setOut(oldOut)
+        System.setErr(oldErr)
+      }
+    }
+
     /** load and run the code using reflection.
      *  @return  A pair consisting of the run's result as a string, and
      *           a boolean indicating whether the run succeeded without throwing
      *           an exception.
      */
-    def loadAndRun(): (String, Boolean) = {
+    def loadAndRun(): (List[String], Boolean) = {
       val interpreterResultObject: Class[_] =
         Class.forName(resultObjectName, true, classLoader)
-      val resultValMethod: java.lang.reflect.Method =
+      val valMethodRes: java.lang.reflect.Method =
         interpreterResultObject.getMethod("result")
       try {
-        (resultValMethod.invoke(interpreterResultObject).toString, true)
+        withOutput(new ByteOutputStream) { ps =>
+          val rawRes    = valMethodRes.invoke(interpreterResultObject).toString
+          val res       =
+            if (List("auto", "always").contains(ictx.settings.color.value))
```

Comment:
Code duplication with `AmmoniteReader.scala`, sounds like you should have a `ictx.useColors` method or something.